content/blog/duckplyr-1-0-0/index.Rmd

```diff
@@ -0,0 +1,209 @@
+* [x] Add intro sentence, e.g. the standard tagline for the package
+* [x] `usethis::use_tidy_thanks()`
+-->
+
+We're very chuffed to announce the release of [duckplyr](https://duckplyr.tidyverse.org) 1.0.0.
+duckplyr is a drop-in, fully compatible replacement for dplyr, powered by [DuckDB](https://duckdb.org/) for speed.
+It joins the rank of dplyr backends together with [dtplyr](https://dtplyr.tidyverse.org) and [dbplyr](https://dbplyr.tidyverse.org).
+You can use it instead of dplyr for data small or large.
+
+<!-- FIXME:
+
+We have many more dplyr backends, the two above are just from the tidyverse.
+GitHub search: https://github.com/search?q=org%3Acran+%2FS3method%5B%28%5D%28mutate%7Csummarise%29+*%2C%2F&type=code
+Do we need an "awesome dplyr" like https://github.com/krlmlr/awesome-vctrs/?
+
+-->
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("duckplyr")
+```
+
+In this article, we'll show how duckplyr can help you with data of different size, explain how you can help improve the package, and ... .
+
+## A drop-in replacement for dplyr
+
+The duckplyr package is a _drop-in replacement for dplyr_ that uses _DuckDB for speed_.
+You can simply _drop_ duckplyr into your pipeline by loading it, then computations will be efficiently carried out by DuckDB.
+DuckDB is a fast in-memory analytical database system[^duckdb].
+
+[^duckdb]: If you haven't heard about it, you can watch [Hannes Mühleisen's keynote at posit::conf(2024)](https://www.youtube.com/watch?v=GELhdezYmP0&feature=youtu.be).
+
+
+```{r}
+library(conflicted)
+library(duckplyr)
+conflict_prefer("filter", "dplyr", quiet = TRUE)
+library(babynames)
+
+out <- babynames |>
+  mutate(prevalence = if_else(prop >= 0.01, "frequent", "rare")) |>
+  summarize(
+    .by = c(sex, year, prevalence),
+    babies_n = sum(n)
+  ) |>
+  filter(sex == "F")
+class(out)
+out
+```
+
+Like with other dplyr backends like dtplyr and dbplyr, duckplyr allows you to get faster results.
+Unlike other dplyr backends, duckplyr does not require you to learn a different syntax.
+
+The duckplyr package is fully compatible with dplyr: if an operation cannot be carried out with DuckDB, it is automatically outsourced to dplyr.
+Over time, we expect fewer and fewer fallbacks to dplyr to be needed.
+
+## How to use duckplyr
+
+To _replace_ dplyr with duckplyr, you can:
+
+- Load duckplyr and then keep your pipeline as is. Calling `library(duckplyr)` overwrites dplyr methods, enabling duckplyr for the entire session no matter how data.frames are created.
+    This is shown in the example above.
+
+- Create individual "duck frames" using _conversion functions_ like `duckdb_tibble()` or `as_duckdb_tibble()`, or _ingestion functions_ like `read_csv_duckdb()`.
+
+
+Then, the data manipulation pipeline uses the exact same syntax as a dplyr pipeline.
+The duckplyr package performs the computation using DuckDB.
+
+```{r}
+# Undo the effect of library(duckplyr)
+methods_restore()
```

Comment:
Could we drop this for the purposes of simplicity?

Reply:
Hidden now

Reply:
In my view, hiding is worse than doing without. Now, with the benchmark, we cannot do without. I propose to show it and talk about it.

Reply:
Added it back.